### DISCUSSION TITLE
Packaging MacOS bundle (captive runtime) using AIR SDK on Windows?

### DISCUSSION THREAD

**Discussion by 15-08-2020** [airsdk/Adobe-Runtime-Support#3597] [General]
Hi!

I'm new to AIR. I'm using VSCode IDE with the AS3MXML extension.
I've successfully built a simple Windows application (captive runtime) using AIR SDK on Windows. I'm using an Asus laptop.
I intend to build a Mac version of the application, then I'll test it using a virtual machine (like VMWare).
But when I switched to target Mac application (also captive runtime), I can still test the app, but I can't build the app. I got this error:

```
The terminal process "...\java.exe '-jar', '...asconfigc.jar', '--sdk', '...\AdobeAir\Windows\51.1', '--debug=false', '--project', '...asconfig.json', '--air', 'bundle', '--verbose=true'" terminated with exit code: 2.
```

What should I do now? Do I have to buy a Mac to do this task?

Any ideas are welcome!

Thank you!

**Comment by ajwfrost** [comment]
Hi

For the desktop platforms, you need to be running on the platform that you're targeting. So for building a macOS bundle, you need to run the AIR SDK on a macOS machine...

This is because, as part of the build process, we have to trigger certain OS-specific tools. So for example on Windows we are editing the resource information within executable files in order to update icons etc; on macOS we are generating icon files and performing code-signing operations, etc.

There are online options for macOS development machines if you don't want to buy a mac, I've not used them myself yet and don't know if it would be cost-effective long term. e.g. search for "cloud service for macos"?

thanks

**Reply by 15-08-2020** [reply]
I got it.
Thank you for helping me!
